README.md

```diff
@@ -0,0 +1,98 @@
+# Activity Streams Add-on
+
+## ActionManager
+When you instantiate an `ActionManager`, you give it a list of types which are valid for the application. If at any time you try to create an action with a type that isn't part of that list, it will throw an error. Yay!
+
+```js
+const am = new ActionManager(["STUFF_REQUEST", "STUFF_RESPONSE"]);
+
+```
+
+You can find the action manager instance for Activity streams at `src/actions/action-manager`.
+
+### Dispatching actions
+To dispatch actions, all you have to do is call `am.actions.YourActionType(...)`.
+
+By default, there is a `Action` type action defined which simply takes an object as an argument. However, it is a good idea to define actions that are specific enough to prevent typos and formatting errors.
+
+```js
+const {actions} = require("actions/action-manager");
+
+// These are all equivalent
+this.dispatch(actions.Action({
+  type:"TOP_FRECENT_SITES_REQUEST",
+  meta: {expect: "TOP_FRECENT_SITES_RESPONSE"}
+}));
+this.dispatch(actions.RequestExpect("TOP_FRECENT_SITES_REQUEST", "TOP_FRECENT_SITES_RESPONSE"));
+this.dispatch(actions.GetTopFrecentSites());
+```
+
+### Defining new actions
+You can define new actions on the `ActionManager` instance with `am.defineActions`. All action definitions should be functions that return a plain object representing the action.
+
+```js
+function Foo(data) {
+  return {type: 'FOO', data};
+}
+am.defineActions({Foo});
+```
+
+All new actions will have some basic validators applied to them like checking for the `type` property. Action functions also get called in the context of the action manager instance.
+
+#### Extending existing actions
+Most new actions will actually be extensions of new ones. To do this, simply call another action function inside your new one.
+
+It is generally **not** a good idea to call `this.actions.SomeAction` inside another action, as that will result in the validators being run twice. Use the plain `SomeAction` function instead, or you can access it from `this.actions.SomeAction.definition`.
+
+```js
+function Request(type, query) {
+  return {type, query};
+}
+function GetFoo() {
+  return Request(foo);
+}
+am.defineActions({Request, GetFoo});
+```
+
+## Conventions for Actions
+In this project we use a modified version of the `Standard Flux Action`. Our standard action looks something like this:
+
+```js
+{
+  // The type should be upper-case.
+  // Request-type actions should end in _REQUEST
+  // Response-type actions should end in _RESPONSE
+  // Required.
+  type: "SOMETHING_REQUEST",
+
+  // Only for response-type actions.
```

Comment:
It might also be for requests. e.g. for search completion, where you'd send the keystrokes so far